sensor/common/centralclient/grpc_connection.go

```diff
@@ -78,6 +78,11 @@ func (f *centralConnectionFactoryImpl) getCentralGRPCPreferences() (*v1.Preferen
 // func finishes.
 // f.okSignal is used if the connection is successful and f.stopSignal if the connection failed to start.
 func (f *centralConnectionFactoryImpl) SetCentralConnectionWithRetries(conn *util.LazyClientConn, certLoader CertLoader) {
+	// This should be called with the okSignal and the stopSignal un-triggered.
+	// If they are triggered, reset them.
+	if f.stopSignal.IsDone() || f.okSignal.IsDone() {
```

Comment:
Since we reset the signals inside the function. Do you think it's still useful to add a log line here?

Reply:
Yeah, it signals that some internal invariant is broken (by some future change) which is possible because someone can set these signals outside the factory I think.

Reply:
If we don't reset the signals outside we will see the warning on each reconnect. I think that logging would make sense if we delegate reseting the signals outside the factory.

I'm also thinking about changing the `StopSignal` and `OkSignal` functions to return read only signals so we make sure the signals are only changed within the factory. WDYT?

Reply:
> we will see the warning on each reconnect.
How come? I thought we don't set *both* atm, `if f.stopSignal.IsDone() && f.okSignal.IsDone() { log...}`

Can we do read-only signals? If so, I think it's a good idea.

Reply:
> Instead, I'd suggest to log a warning if both signals are set

🤦 I need to read better. I though you suggested *any* signal not *both*. Yeah both should not be set at the same time, make sense to add the warning if for whatever reason in the future the code is changed and we run in this situation again.

> Can we do read-only signals?

For `ErrorSignal` we can. For `Signal` we do not, but it'd easy to add the interface.